processor/src/tasks/processReadme.js

```diff
@@ -0,0 +1,127 @@
+const globule = require('globule')
+const path = require('path')
+const superagent = require('superagent')
+const cheerio = require('cheerio')
+
+const { exists, readFile, writeFile } = require('../utils')
+const { GITEA_URL } = require('../env')
+
+function processReadme(events, inputDir, kitspaceYaml, outputDir, projectFullname) {
+  if (kitspaceYaml.multi) {
+    const projectNames = Object.keys(kitspaceYaml.multi)
+    return Promise.all(
+      projectNames.map(projectName => {
+        const projectOutputDir = path.join(outputDir, projectName)
+        const projectKitspaceYaml = kitspaceYaml.multi[projectName]
+
+        return _processReadme(
+          events,
+          inputDir,
+          projectKitspaceYaml,
+          projectOutputDir,
+          projectFullname,
+        )
+      }),
+    )
+  }
+
+  return _processReadme(events, inputDir, kitspaceYaml, outputDir, projectFullname)
+}
+
+async function _processReadme(
+  events,
+  inputDir,
+  kitspaceYaml,
+  outputDir,
+  projectFullname,
+) {
+  const readmePath = path.join(outputDir, 'readme.json')
+
+  events.emit('in_progress', readmePath)
+
+  if (await exists(readmePath)) {
+    events.emit('done', readmePath)
+    return
+  }
+
+  let readmeInputPath
+
+  if (kitspaceYaml.readme) {
+    readmeInputPath = path.join(inputDir, kitspaceYaml.readme)
+  } else {
+    const readmeFile = findReadmeFile(inputDir)
+    if (readmeFile === null) {
+      events.emit('failed', readmePath, "couldn't find readme file")
+      return
+    }
+    readmeInputPath = readmeFile
+  }
+
+  const rawMarkdown = await readFile(readmeInputPath, { encoding: 'utf8' })
+  const readmeAsHTML = await renderMarkdown(rawMarkdown)
+
+  const renderedReadme = postProcessMarkdown(readmeAsHTML, projectFullname)
+
+  await writeFile(readmePath, JSON.stringify({ body: renderedReadme }))
+    .then(() => events.emit('done', readmePath))
+    .catch(e => events.emit('failed', readmePath, e))
+}
+
+function findReadmeFile(inputDir) {
+  const readmeFiles = globule.find(
+    path.join(inputDir, 'readme?(.markdown|.mdown|.mkdn|.md|.rst)'),
+    {
+      nocase: true,
+    },
+  )
+
+  if (readmeFiles.length >= 1) {
+    return readmeFiles[0]
+  } else {
+    return null
+  }
+}
+
+async function renderMarkdown(rawMarkdown) {
+  const giteaMarkdownEndpoint = `${GITEA_URL}/api/v1/markdown/raw`
```

Comment:
Instead of accessing it throught the external domain we can go through docker networking, i.e. 

```
  const giteaMarkdownEndpoint = `http://gitea:3000/api/v1/markdown/raw`
```

Reply:
Resolved in c0f4757.